zokrates_core/src/flatten/mod.rs

```diff
@@ -1225,19 +1194,9 @@ impl<'ast, T: Field> Flattener<'ast, T> {
 
                 // Handle complex parameters and assign values:
                 // Rename Parameters, assign them to values in call. Resolve complex expressions with definitions
-                // Clippy doesn't like the fact that we're collecting here, however not doing so leads to a borrow issue
```

Comment:
after some refactoring I did not encounter this issue explained in the comment? @Schaeff have a look just to be sure I'm not missing something here

Reply:
I feel it was a clippy bug anyway!